services/galley/src/Galley/Cassandra/TeamFeatures.hs

```diff
@@ -140,13 +143,56 @@ setSelfDeletingMessagesStatus tid status = do
           <> ", self_deleting_messages_ttl) "
           <> "values (?, ?, ?)"
 
+setPaymentStatus ::
+  forall (a :: TeamFeatureName) m.
+  ( MonadClient m,
+    HasPaymentStatusCol a
+  ) =>
+  Proxy a ->
+  TeamId ->
+  PaymentStatus ->
+  m PaymentStatus
+setPaymentStatus _ tid (PaymentStatus paymentStatus) = do
+  retry x5 $ write insert (params LocalQuorum (tid, paymentStatus))
+  pure (PaymentStatus paymentStatus)
+  where
+    insert :: PrepQuery W (TeamId, PaymentStatusValue) ()
+    insert =
+      fromString $
+        "insert into team_features (team_id, " <> paymentStatusCol @a <> ") values (?, ?)"
+
+getPaymentStatus ::
+  forall (a :: TeamFeatureName) m.
+  ( MonadClient m,
+    MaybeHasPaymentStatusCol a
+  ) =>
+  Proxy a ->
+  TeamId ->
+  m (Maybe PaymentStatus)
```

Comment:
```suggestion
  m (Maybe PaymentStatusValue)
```

It's the same thing, but I think I've just had to deconstruct the `PaymentStatus` into a `PaymentStatusValue` somewhere after calling this.  (I may be wrong, it's late.  You decide.)